Core/Src/main.c

```diff
@@ -67,8 +69,8 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
-  uint8_t *rx_buf = malloc(2);
-  uint8_t tx_buf = 1; //transmitted to ADC to initiate read from analog source
+  uint8_t rx_buf[2] = {0};
+  uint8_t tx_buf = ACTIVATE_ADC_VALUE; //transmitted to ADC to initiate read from analog source
```

Comment:
I believe that this also needs to select the single/double mode and ADC channel to read. And looking in more depth, the buffer length needs to be longer, even though only 10 bits of data are returned to synchronize the duplex communication. There is a diagram in I believe section 5 or 6 of the ADC datasheet that shows the timing of communication, and a table for the channel and single selection commands.